src/DynamoCore/Graph/Workspaces/UndoRedo.cs

```diff
@@ -34,15 +34,35 @@ internal UndoRedoRecorder UndoRecorder
         /// </summary>
         public bool CanUndo
         {
-            get { return ((null != undoRecorder) && undoRecorder.CanUndo); }
+            get
+            {
+                if (UndoRedoLocked)
+                {
+                    return false;
+                }
+                return ((null != undoRecorder) && undoRecorder.CanUndo);
+            }
         }
 
         /// <summary>
         ///     Determine if redo operation is currently possible.
         /// </summary>
         public bool CanRedo
         {
-            get { return ((null != undoRecorder) && undoRecorder.CanRedo); }
+            get
+            {
+                if (UndoRedoLocked)
+                {
+                    return false;
+                }
+                return ((null != undoRecorder) && undoRecorder.CanRedo);
+            }
+        }
+
+        internal bool UndoRedoLocked
```

Comment:
Usually for any bool typed property, we try to follow the naming pattern `IsXXX`, so in this case I would recommend `IsUndoRedoLocked`